From 32b11a24162542deeaa6bc75894ec4b07186849d Mon Sep 17 00:00:00 2001
From: arunthomas <arunthomasb@ami.com>
Date: Fri, 5 Apr 2024 18:29:00 +0530
Subject: [PATCH 241/241] Rearrage the Order of verifying the Version.

populateSoftwareInformation() wil be called from BMCWEB handlers
bios.hpp, systems.hpp and managers.hpp to get the version of
firmware running is bios and bmc. But besides bios and bmc there are
object path like "/xyz/openbmc_project/software/rot_fw_active" and
"/xyz/openbmc_project/software/me" with the same interface. When we
validate the version, before comparing the Purpose this leads to iternal
error since "/xyz/openbmc_project/software/me" has Version as "".
So if we rearrange the order for validation this will get fixed untill we
provide the swVersionPurpose for me which is not used from BMCWEB so far.

Tested: Verified the response for Systems, Managers Instance and Bios URI
is success with status code 200.

Signed-off-by: arunthomas <arunthomasb@ami.com>
---
 redfish-core/include/utils/sw_utils.hpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/redfish-core/include/utils/sw_utils.hpp b/redfish-core/include/utils/sw_utils.hpp
index 8a715e67..011d17a1 100644
--- a/redfish-core/include/utils/sw_utils.hpp
+++ b/redfish-core/include/utils/sw_utils.hpp
@@ -174,17 +174,17 @@ inline void populateSoftwareInformation(
                         return;
                     }
 
-                    if (version == nullptr || version->empty())
-                    {
-                        messages::internalError(asyncResp->res);
-                        return;
-                    }
                     if (swInvPurpose == nullptr ||
                         *swInvPurpose != swVersionPurpose)
                     {
                         // Not purpose we're looking for
                         return;
                     }
+                    if (version == nullptr || version->empty())
+                    {
+                        messages::internalError(asyncResp->res);
+                        return;
+                    }
 
                     BMCWEB_LOG_DEBUG("Image ID: {}", swId);
                     BMCWEB_LOG_DEBUG("Running image: {}", runningImage);
-- 
2.39.2

